akka-http/src/main/scala/akka/http/javadsl/server/directives/WebSocketDirectives.scala

```diff
@@ -66,6 +68,8 @@ abstract class WebSocketDirectives extends SecurityDirectives {
    * the request is rejected with an [[UnsupportedWebSocketSubprotocolRejection]] rejection.
    *
    * To support several subprotocols you may chain several `handleWebSocketMessagesForOptionalProtocol` routes.
+   *
+   * To support any subprotocol `*` can be used.
```

Comment:
```
`"*"` can be used 
```

Reply:
It's a bit unfortunate that we have `Optional[String]` and not a select function here hm...

Reply:
Actually I think passing in the selected subprotocol as a parameter rather than a function here makes sense, since you have to have to provide a `Flow` which implements the subprotocol you select, so subprotocol selection must be done before creating the `Flow` anyway.